docs/photoniq/faas/faas-cli-commands.md

```diff
@@ -0,0 +1,437 @@
+---
+sidebar_position: 11
+title: FaaS CLI Commands
+---
+
+The PhotonIQ FaaS Command Line Interface (CLI) is a command line interface that enables you to manage local and remote functions.
+
+Here's the complete CLI documentation formatted as markdown, including all the provided commands and options:
+
+## Local Commands
+
+Use local commands to create and manage functions locally.
+
+### faas new
+
+Create a new local function with the specified language.
+
+**Syntax:**
+
+```bash
+faas new <local-fn-name> [--lang rust|js]
+```
+
+**Default:**
+
+- `--lang rust`
+
+**Examples:**
+
+```bash
+  # Create a new function in Rust
+  faas new "my-rust-function"
+
+  # Create a new function in JavaScript
+  faas new "my-js-function" --lang js
+```
+
+**Options:**
+
+```bash
+  -h, --help           Help for the new command.
+  --lang <rust|js>     Programming language for the function (default: rust).
+```
+
+### faas build
+
+Build a specified local function.
+
+**Syntax:**
+
+```bash
+faas build <local-fn-name>
+```
+
+**Examples:**
+
+```bash
+  # Build a local function
+  faas build "my-function"
+```
+
+**Options:**
+
+```bash
+  -h, --help           Help for the build command.
+```
+
+### faas list
+
+List all local functions.
+
+**Syntax:**
+
+```bash
+faas list
+```
+
+**Examples:**
+
+```bash
+  # List all local functions
+  faas list
+```
+
+**Options:**
+
+```bash
+  -h, --help           Help for the list command.
+```
+
+### faas info
+
+Get information about a local function.
+
+**Syntax:**
+
+```bash
+faas info <local-fn-name>
+```
+
+**Examples:**
+
+```bash
+  # Get information about a local function
+  faas info "my-function"
+```
+
+**Options:**
+
+```bash
+  -h, --help           Help for the info command.
+```
+
+### faas execute
+
+Execute a local function using the specified data file.
+
+**Syntax:**
+
+```bash
+faas execute <local-fn-name> --data <data-file>
+```
+
+**Examples:**
+
+```bash
+  # Execute a local function with data from a JSON file
+  faas execute "my-function" --data data.json
+```
+
+**Options:**
+
+```bash
+  -h, --help           Help for the execute command.
+  --data <data-file>   JSON file containing data to be passed to the function.
```

Comment:
@dlozina-macrometa - We have improved the command. Was changed `--data` to `--file` and added  `--data` which contains data as a string in command:

```
--file <data-file>   JSON file containing data to be passed to the function.
--data '<your-data>'  data to be passed to the function.
```

Reply:
Also was added optional parameter `--header`, `-H` 
```
--header 'key1: value1'  HTTP header to be passed to the function.
```